File: scala-hello/src/main/scala/FizzBuzz.scala
/*
 * Copyright 2017-2019 lxb.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
object FizzBuzz extends App {
  var n = args(0).toInt;
  for (a <- 1 to n) {
    if (a % 3 == 0 && a % 5 == 0) {
      println("FizzBuzz")
    } else if (a % 3 == 0) {
      println("Fizz")
    } else if (a % 5 == 0) {
      println("Buzz")
    } else {
      println(a)
    }
  }
}

File: clojure-hello/src/main/clojure/FizzBuzz.clj
;
; Copyright 2017-2023 lxb.
;
; Licensed under the Apache License, Version 2.0 (the "License");
; you may not use this file except in compliance with the License.
; You may obtain a copy of the License at
;
;      http://www.apache.org/licenses/LICENSE-2.0
;
; Unless required by applicable law or agreed to in writing, software
; distributed under the License is distributed on an "AS IS" BASIS,
; WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
; See the License for the specific language governing permissions and
; limitations under the License.
;
(ns FizzBuzz
      (:gen-class))

(defn -main [& args] 
  (def n (Integer/parseInt (first args)))
  (loop [x 1]
    (when (<= x n)
      (println 
        (cond
          (and (= (mod x 3)0) (= (mod x 5) 0)) "FizzBuzz"
          (=(mod x 3 ) 0) "Fizz"
          (=(mod x 5 )0 ) "Buzz"
          :else x
        )
      )
    (recur (+ x 1))))
)